source/common/stats/symbol_table_impl.h

```diff
@@ -465,6 +364,52 @@ class StatNameTempStorage : public StatNameStorage {
   SymbolTable& symbol_table_;
 };
 
+// Represenets an ordered container of StatNames. The encoding for each StatName
+// is byte-packed together, so this carries less overhead than allocating the
+// storage separately. The tradeoff is there is no random access; you can only
+// itereate through the StatNames.
+class StatNameList {
+public:
+  ~StatNameList();
+
+  /**
+   * Populates the StatNameList from a list of encodings. This is not done at
+   * construction time to enable StatNameList to be instantiated directly in
+   * a class that doesn't have a live SymbolTable when it is constructed.
+   *
+   * @param encodings The list names to encode.
+   * @param symbol_table The symbol table in which to encode the names.
+   */
+  void populate(const std::vector<absl::string_view>& encodings, SymbolTable& symbol_table);
```

Comment:
How/who is using this? I dig the explanation in the comment, but would like to see how it is used in practice.

Reply:
MetricImpl uses it in #4980 . Prior to the latest update this impl was basically inlined into MetricImpl in that PR, now it's dependent on StatNameList, which is a lot nicer as the deep knowledge of StatName semantics stays in one place.